server/public/pluginapi/i18n/i18n_test.go

```diff
@@ -144,9 +144,6 @@ func TestLocalizeDefaultMessage(t *testing.T) {
 			},
 		})
 		api.On("GetBundlePath").Return(".", nil)
-		api.On("LogWarn", mock.AnythingOfType("string"),
-			mock.AnythingOfType("string"), mock.AnythingOfType("string"),
-			mock.AnythingOfType("string"), mock.AnythingOfType("string")).Return()
```

Comment:
@KuSh is there a behavioral change here? I think empty string does gets localized somehow with the `v2`. Is this accurate? If so, we are safe to ignore these logs. Though, now this would become unnoticeable.

https://github.com/nicksnyder/go-i18n/blob/85a3f2b18e445306e0a1b6076d4f2a48ae32f066/v2/i18n/localizer.go#L158

Reply:
I don't know I'll check how and why, but this can at list be handled in LocalizeDefaultMessage if needed

Reply:
@isacikgoz This is the actual behavior of go-i18n v2 for sure but I'm not certain what you're asking me.

This code was not part of my PR and I didn't touch it nor does it use things I've changed (mostly in shared/i18n) so I don't know what was the previous behavior you're talking about.

I saw this was added in august ([3ee5432](https://github.com/mattermost/mattermost/commit/3ee5432664d8530390b4216b61fe1a4afd274dd3)) but my PR has been around since october last year.

If pluginapi was previously using go-i18n v1 it was a lot different. I've tried to check its code but without knowing how it was done I can't really be sure. I didn't found any specific behavior around empty messages in https://github.com/nicksnyder/go-i18n/blob/3c6ee9071cf061f4b0101e14412f74d6e7a8ada6/i18n/bundle/bundle.go#L342 nor https://github.com/nicksnyder/go-i18n/blob/3c6ee9071cf061f4b0101e14412f74d6e7a8ada6/i18n/bundle/bundle.go#L303

If empty messages needs to be noticed we can add a check for an empty string after calling `LocalizeMessage(m)` in `LocalizeDefaultMessage`

On a side note, while checking this, I also saw that you're using a relatively old version of go-i18n v2, perhaps it would be a good idea to bump it in this PR. What do you think?

Reply:
Ok so after digging more it is clearly a behavioral change, certainly due to https://github.com/nicksnyder/go-i18n/pull/189

In v2.0.3 `getTemplate(messageID, lc.DefaultMessage)` was returning nil so a `MessageNotFoundErr{messageID: messageID}` error was returned, now getMessageTemplate does not return an error if defaultMessage is not nil but just empty

I'll make a PR to go-i18 to change that behavior. In the mean time we can also add a special case in LocalizeDefaultMessage if final string is empty. Just let me know if it suits you

Reply:
I think bumping the go-i18n v2 is good idea, at least have a consistency would be enough. I think it would be wise to return an error if the string is empty, otherwise it's just silently ignoring a fact. I'm okay with the modifying `LocalizeDefaultMessage` for now.

Reply:
I've pushed a PR to go-i18n but it is not merged ATM and will require a release so I'll bump to latest available and add a check with a comment for now 👍

Reply:
Awesome!

Reply:
Ok so my PR on go-i18n v2 has been merged and a version has been released so I'll take that into account here !